crates/collab/src/tests/integration_tests.rs

```diff
@@ -3412,7 +3412,8 @@ async fn test_local_settings(
     });
 }
 
-#[gpui::test(iterations = 10)]
+// TODO(#21034): This test is flaky for some execution orders. Set iterations back to 10 once fixed.
```

Comment:
I think we'd prefer to fix this bug as part of this PR, rather than put a TODO on it.

Reply:
Yeah, should probably just do the fix!  Unfortunately it may be fairly involved, as it requires the effects of `proto::UpdateBufferFile` and `proto::BufferSaved` to be applied together rather than separately.  I believe the effects of this would be hard to see in the UI since they are emitted at the same time.